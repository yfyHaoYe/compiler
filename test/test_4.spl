int test_1_r09()
{
  int m = 1;
  float n = 2.2;
  float f = 0.2;
  int arr[10], kfc[50];
  arr[m] = 7;
  kfc[n] = 8;
  n = n - f;
  kfc[n] = 50;
  return kfc[2];
}