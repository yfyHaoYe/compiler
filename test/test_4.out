Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test_1_r09
        LP
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: m
                ASSIGN
                Exp (3)
                  INT: 1
            SEMI
          DefList (4)
            Def (4)
              Specifier (4)
                TYPE: float
              DecList (4)
                Dec (4)
                  VarDec (4)
                    ID: n
                  ASSIGN
                  Exp (4)
                    FLOAT: 2.2
              SEMI
            DefList (5)
              Def (5)
                Specifier (5)
                  TYPE: float
                DecList (5)
                  Dec (5)
                    VarDec (5)
                      ID: f
                    ASSIGN
                    Exp (5)
                      FLOAT: 0.2
                SEMI
              DefList (6)
                Def (6)
                  Specifier (6)
                    TYPE: int
                  DecList (6)
                    Dec (6)
                      VarDec (6)
                        VarDec (6)
                          ID: arr
                        LB
                        INT: 10
                        RB
                    COMMA
                    DecList (6)
                      Dec (6)
                        VarDec (6)
                          VarDec (6)
                            ID: kfc
                          LB
                          INT: 50
                          RB
                  SEMI
        StmtList (7)
          Stmt (7)
            Exp (7)
              Exp (7)
                Exp (7)
                  ID: arr
                LB
                Exp (7)
                  ID: m
                RB
              ASSIGN
              Exp (7)
                INT: 7
            SEMI
          StmtList (8)
            Stmt (8)
              Exp (8)
                Exp (8)
                  Exp (8)
                    ID: kfc
                  LB
                  Exp (8)
                    ID: n
                  RB
                ASSIGN
                Exp (8)
                  INT: 8
              SEMI
            StmtList (9)
              Stmt (9)
                Exp (9)
                  Exp (9)
                    ID: n
                  ASSIGN
                  Exp (9)
                    Exp (9)
                      ID: n
                    MINUS
                    Exp (9)
                      ID: f
                SEMI
              StmtList (10)
                Stmt (10)
                  Exp (10)
                    Exp (10)
                      Exp (10)
                        ID: kfc
                      LB
                      Exp (10)
                        ID: n
                      RB
                    ASSIGN
                    Exp (10)
                      INT: 50
                  SEMI
                StmtList (11)
                  Stmt (11)
                    RETURN
                    Exp (11)
                      Exp (11)
                        ID: kfc
                      LB
                      Exp (11)
                        INT: 2
                      RB
                    SEMI
        RC
